### PR TITLE
More REST generator progress

### PR DESCRIPTION
**Pull request by jskeet**

### CODE REVIEW

File: Google.Api.Generator.Rest/Models/MethodModel.cs [hunk 1]
```diff
@@ -15,25 +15,84 @@
...
+        public Typ RequestTyp { get; }
         public string Name { get; }
 
-        public MethodModel(PackageModel package, string name, RestMethod restMethod)
+        public MethodModel(PackageModel package, ResourceModel resource, string name, RestMethod restMethod)
         {
             Package = package;
+            Resource = resource;
             Name = name.ToUpperCamelCase();
+            ParentTyp = resource?.Typ ?? package.ServiceTyp;
```
Comment (amanda-tarafa) [inline Google.Api.Generator.Rest/Models/MethodModel.cs:46]: Oh, I din't know that the methods could be defined outside of the resources, directly on the service, don't remember having seen that.

File: Google.Api.Generator.Rest/Models/MethodModel.cs [hunk 2]
```diff
@@ -15,25 +15,84 @@
...
+        public IReadOnlyList<ParameterModel> Parameters { get; }
+        public Typ ParentTyp { get; }
+        public Typ RequestTyp { get; }
         public string Name { get; }
 
-        public MethodModel(PackageModel package, string name, RestMethod restMethod)
+        public MethodModel(PackageModel package, ResourceModel resource, string name, RestMethod restMethod)
         {
             Package = package;
+            Resource = resource;
```
Comment (amanda-tarafa) [inline Google.Api.Generator.Rest/Models/MethodModel.cs:44]: I think you are not using this anywhere else, I'd say, remove it, if it's not relevant for MethodModel then it shouldn't be relevant outside, or if it becomes relevant then maybe MethodModel should encapsulate that.

File: Google.Api.Generator.Rest/Models/MethodModel.cs [hunk 3]
```diff
@@ -15,25 +15,84 @@
...
+        }
+
+        private static IReadOnlyList<ParameterModel> CreateParameterList(PackageModel package, RestMethod restMethod)
+        {
+            var parameterOrder = restMethod.ParameterOrder ?? new List<string>();
+            // TODO: Skip the Alt parameter
+            // TODO: Handle the body parameter if restMethod.Request is non-null
+            return (restMethod.Parameters ?? new Dictionary<string, JsonSchema>())
+                .Select(pair => new ParameterModel(pair.Key, pair.Value))
+                .OrderBy(p => !p.IsRequired)
```
Comment (amanda-tarafa) [inline Google.Api.Generator.Rest/Models/MethodModel.cs:59]: I think you can directly skip here the ones that are not required. Or are you planning to be able to generate overloads later?

**Review by amanda-tarafa — APPROVED** [review]
LGTM, just a couple of comments.

File: Google.Api.Generator.Rest/Models/MethodModel.cs (hunk 1 above)
Comment (jskeet) [reply Google.Api.Generator.Rest/Models/MethodModel.cs:46]: I don't have any concrete examples of it at the moment... we might be able to remove it later.

File: Google.Api.Generator.Rest/Models/MethodModel.cs (hunk 2 above)
Comment (jskeet) [reply Google.Api.Generator.Rest/Models/MethodModel.cs:44]: I'm generally being fairly lax about what to define as properties. It makes things a lot quicker for development. I suggest that we don't worry about it for the moment, but that we go back over all the models later to see what we actually need to expose. It's definitely inconsistent right now.

File: Google.Api.Generator.Rest/Models/MethodModel.cs (hunk 3 above)
Comment (jskeet) [reply Google.Api.Generator.Rest/Models/MethodModel.cs:59]: Not overloads, but they become properties on the request class - we'll definitely want them.

**Comment by jskeet** [comment]
(Merging as-is.)

**Pull request merged**